src/mkdir.js

```diff
@@ -75,7 +75,7 @@ function _mkdir(options, dirs) {
 
     try {
       if (options.fullpath) {
-        mkdirSyncRecursive(dir);
+        mkdirSyncRecursive(path.resolve(dir));
```

Comment:
Yes, that's correct.

It's needed here because, for each recursive call, we process another piece of the path. This can lead us to processing something that isn't valid for `mkdir`, like `.` or `..`. It can also lead us to processing the same directory twice (ex. `foo/../foo/`) which also isn't valid.